service/src/main/java/org/mskcc/cmo/metadb/service/impl/PatientServiceImpl.java

```diff
@@ -36,5 +36,16 @@ public MetadbPatient getPatientByCmoPatientId(String cmoPatientId) {
     public UUID getPatientIdBySample(UUID metadbSampleId) {
         return patientRepository.findPatientIdBySample(metadbSampleId);
     }
+    
+    @Override
+    public MetadbPatient updateCmoPatientId(String oldCmoPatientId, String newCmoPatientId) {
+        if (getPatientByCmoPatientId(oldCmoPatientId) == null) {
+            return null;
+        }
+        //only update the patient if it is found, hence not null
+        patientRepository.updateCmoPatientIdInPatientNode(oldCmoPatientId, newCmoPatientId);
```

Comment:
this method is returning an instance of `MetadbPatient` already isn't it? So we could just assign it to a variable and return the variable instead of calling this again  in the return statement `getPatientByCmoPatientId(newCmoPatientId)`